Automattic/wp-calypso
# FSE: Add script to automatically sync changes to your wpcom sandbox

### Changes proposed in this Pull Request
* Adds a script to rsync FSE to your sandbox when there are changes
* Allows you to pass `--sync` as an arg to the dev and build commands in order to sync to your sandbox

### SSH config
A prerequisite for this script is being able to reference a named SSH config to sync changes to your sandbox. 

`vi ~/.ssh/config` reveals this entry for me with some details renamed:

```
# If you don't add your keys to the SSH agent, it'll continuously ask you for your password!
Host *
    AddKeysToAgent yes
    IdentityFile ~/.ssh/id_rsa

Host wpcom-sandbox
    User yoursandboxusernamehere
    HostName yoursandboxurlhere.word press.com
```

I figure it's likely that many of us already have this set up. All that you'll have to do is add an entry for `wpcom-sandbox` pointing at your sandbox, and then the script should work without any further setup. Way better than all the pain our new devs have had to go through getting unison set up!

### Test it out 
First, make sure you are **not** running any existing unison or build processes.

1. Make sure you have the above SSH config file set up.
2. `svn up` on your sandbox to get the latest changes. Also note that rsync should work more quickly if you have an active SSH session to your sandbox running at the same time
2. Run `nvm use && npm i` in the root of wp-calypso to get everything up to date.
3. Run `npx lerna run dev --scope='@automattic/full-site-editing' --stream -- -- --sync`

Obviously, the above command is pretty nasty because you have to pipe the args through npx, lerna, npm, and into a node script. I would suggest making an alias for this command in your `~/.bash_profile`.

Once this is running, you should see something like `Running the following commands: dev:*,wpcom-sync` near the bottom of the output.

To test that it's working:
1. Make a modification to a PHP file like `full-site-editing-plugin.php`. Then in your sandbox, open up `wp-content/plugins/full-site-editing-plugin/dev/full-site-editing-plugin.php` and make sure you see the change reflected.
2. Make a modification to a JS file (something you can see in the UI). Then, with a site and the API sandboxed, visit the FSE editor for that site. Make sure you see that change in the UI. (Also verify in the console that you are loading the `dev` version.)

### To do:
- [ ] Display output of sync and build. Currently it seems to be hidden.
